feat(SpawnPlateauProvider): derive target height from elevation and sea level facets

skaldarnar opened the pull request:
With this PR we derive the `targetHeight` of the spawn plateau from the `ElevationFacet` and `SeaLevelFacet` instead of using a fixed value. Might still look funny in some occasions, but the huge craters are gone.

Seed `iXpHtY7fk97YP9VI` would spawn in the water if the sea level would not be considered. 💧 

Seed `WtR9q4zA6yizPN5G` is just a pretty nice spawn point 🙃

skaldarnar committed:
feat(SpawnPlateauProvider): derive target height from elevation and sea level facets

skaldarnar committed:
chore: omit target height when creating PlateauProvider in default world gen

4Denthusiast commented:
If values other than the usual ones (spawn point (0,0), not too large radius) are passed to the constructor, it may fail to generate sometimes because the centre position will be outside of the region where the ElevationFacet is generated. I don't think there's any particularly neat way to fix this, but it would be good to at least check whether the elevation at the centre is available, and give up if not. That way, the severity of this bug would be reduced from an exception to just an odd worldgen feature.

skaldarnar committed:
use common constant for spawn location

jdrueckert commented:
@4Denthusiast How would you rate the probability of running into this?

skaldarnar commented:
Do we have a "graceful" way to give up early on? I can just do an early return, which will skip the facet modification for that region, I guess.

4Denthusiast commented:
> @4Denthusiast How would you rate the probability of running into this?

It's unlikely to actually come up, but if the provider's constructor has these arguments, it should work correctly with values of the arguments other than the usual ones. Another option would be to remove the radius arguments, giving those fixed values instead, and setting the border according to those fixed values.

skaldarnar commented:
Sounds reasonable enough for our use case - will do those changes for the fixed radius and updated border 👍

skaldarnar committed:
feat: add new SpawnPlateauProvider to replace PlateauProvider

I decided to make this a new provider as this is basically a complete change in the way it works - you only give it a position, the rest is fixed within the provider.

skaldarnar committed:
chore: deprecate PlateauProvider

skaldarnar committed:
chore: switch to SpawnPlateauProvider for default world gens

jdrueckert commented on src/main/java/org/terasology/core/world/generator/facetProviders/SpawnPlateauProvider.java:
```diff
@@ -0,0 +1,77 @@
+/**
+ * Flattens the surface in a circular area around a given coordinate.
+ * <p>
+ * The area outside this area will be adjusted up to a fixed radius of {@link SpawnPlateauProvider#OUTER_RADIUS} to
```
"the area outside this area" is a bit confusing

jdrueckert reviewed (approved):
Looks good from my side. Would appreciate a second opinion from @4Denthusiast  though.
